cmd/genji/shell/command.go

```diff
@@ -92,16 +101,97 @@ func runIndexesCmd(db *genji.DB, in []string) error {
 	return fmt.Errorf("usage: .indexes [tablename]")
 }
 
-// runHelpCmd display all available dot commands.
-func runHelpCmd() error {
-	var keys []string
-	for k := range commands {
-		keys = append(keys, k)
+// displayHelp show all available cmd.
+func displayHelp(db *genji.DB, cmd []string) error {
+	for _, c := range commands {
+		// spaces for help indentation and alignment
+		spaces := 25
+		indent := spaces - len(c.Options) - len(c.Name)
+		fmt.Printf("%s %s", c.Name, c.Options)
+		fmt.Printf("%*s%s\n", indent, "", c.Help)
+	}
+
+	return nil
+}
+
+// exitCommand exits this program.
+func exitCommand(db *genji.DB, cmd []string) error {
+	os.Exit(0)
+	return nil
+}
+
+// registerCommands add all cmd and sort them by name.
+func registerCommands() {
+	commands = append(commands, Command{Name: ".indexes",
+		Options: "[table name]",
+		Help:    "Display all indexes or the indexes of the given table name.",
+		Action:  runIndexesCmd})
+
+	commands = append(commands, Command{Name: ".tables", Help: "List names of tables.", Action: displayTables})
+	commands = append(commands, Command{Name: ".help", Help: "Show available cmd.", Action: displayHelp})
+	commands = append(commands, Command{Name: ".exit", Help: "Exit this program.", Action: exitCommand})
+
+}
+
+// getAllCommands returns all cmd.
+func getAllCommands() []string {
+	cmdList := make([]string, len(commands))
+	for i := range commands {
+		cmdList[i] = commands[i].Name
+	}
+
+	return cmdList
+}
+
+// getCommandSuggestion returns the suggested command.
+func getCommandSuggestion(in string) Command {
```

Comment:
This doesn't do what the comment suggests, it creates a command that itself determines which command to run. I think this is a bit hacky.